tests/gitgitgadget.test.ts

```diff
@@ -246,24 +246,31 @@ to have included in git.git [https://github.com/git/git].`);
     mails.splice(0);
     expect(await patches2.generateAndSend(logger, send, undefined,
                                           pullRequestURL))
-        .toEqual("pull.1.v2.git.gitgitgadget@example.com");
+        .toMatch(/pull\.1\.v2\.git\.\d+\.gitgitgadget@example\.com/);
     expect(mails.length).toEqual(5);
     if (await gitCommandExists("range-diff", repo.workDir)) {
         expect(mails[0]).toMatch(/Range-diff vs v1:\n[^]*\n -: .* 4: /);
     }
     expect(await repo.revParse("pr-1/somebody/master-v2")).toBeDefined();
 
-    expect(await notes.get(pullRequestURL)).toEqual({
+    const seriesMeta = await notes.get<IPatchSeriesMetadata>(pullRequestURL);
+    expect(seriesMeta.coverLetterMessageId).not.toBeNull();
```

Comment:
I fear that my advice was not good... The Typescript compiler [complains](https://dev.azure.com/gitgitgadget/git/_build/results?buildId=18091&view=logs&jobId=12f1170f-54f2-53f3-20dd-22fc7dff55f9&taskId=9c939e41-62c2-5605-5e05-fc3554afc9f5&lineStart=39&lineEnd=45&colStart=1&colEnd=1). Maybe it works if you use `.not.toBeUndefined()`?

I fear that you still have to append `!` to the subsequent uses of `seriesMeta.coverLetterMessageId`...

Reply:
Due to the problem raised in issue #130 I did not catch the error before the CI build.  I changed to extract the values to new vars to get past the problem.  Will look at using !.

Reply:
Done ;-)

I hope you did not continue working on that, if so: my apologies for wasting your time!